### PR TITLE
chexmix: new package @0.52.public

### PR DESCRIPTION
**Pull request by A-N-Other**
Adding the `chexmix` package. Also including a convenience wrapper that points `java` at the `.jar` file for the user.

### CODE REVIEW

**Comment by A-N-Other** [comment]
@spackbot fix style

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/chexmix/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+class Chexmix(Package):
+    """ChExMix: the ChIP-exo mixture model. ChExMix aims to characterize protein-DNA binding
+    subtypes in ChIP-exo experiments."""
+
+    homepage = "http://mahonylab.org/software/chexmix"
+    url = "https://github.com/seqcode/chexmix/releases/download/v0.52/chexmix.v0.52.public.jar"
+
+    version(
+        "0.52.public",
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/chexmix/package.py]: One question before merging: why using the `public` suffix? Can we move the suffix to the package name?

Comment (A-N-Other) [reply var/spack/repos/builtin/packages/chexmix/package.py]: Although the [tags](https://github.com/seqcode/chexmix/tags) don't include `public`, the `.jar` files are either generated with it (`0.52`, `0.41`) or without (`0.51`) it in the filename. Can't see any obvious reason ...

Comment (alalazo) [reply var/spack/repos/builtin/packages/chexmix/package.py]: If there are no differences, let's just use:
```suggestion
        "0.52",
```
?

Comment (A-N-Other) [reply var/spack/repos/builtin/packages/chexmix/package.py]: Can do, but then we'd need to specify a manual `url=` for `0.52` and any other versions for which the download matches `*.public.jar`.
```
# spack versions chexmix
==> Safe versions (already checksummed):
  0.52.public
==> Remote versions (not yet checksummed):
  0.51  0.41.public
# spack checksum chexmix 0.52
==> Error: Could not find any remote versions for chexmix
```
Fine with either - which do you prefer?

Comment (alalazo) [reply var/spack/repos/builtin/packages/chexmix/package.py]: Let's give https://github.com/seqcode/chexmix/issues/6 some time. If there's a reply we might solve both issues.

Comment (A-N-Other) [reply var/spack/repos/builtin/packages/chexmix/package.py]: Just looping back to some of my older PRs, @alalazo.

Silence there for now, so what's your preference here - leave as is (version searches work, but some will be end up being `.public` suffixed) or manually specify urls for the different versions?

Comment (alalazo) [reply var/spack/repos/builtin/packages/chexmix/package.py]: Manually specified urls: more maintenance work, but also saner for end users

Comment (A-N-Other) [reply var/spack/repos/builtin/packages/chexmix/package.py]: Have switched over to manual urls. Bit more fiddly than just that as we also need control over the name of the downloaded file for the `install()` process, so have added a `run_before()` decorator to handle that.

**Review by alalazo — APPROVED** [review]

**Pull request merged**